tests/net/socket/getaddrinfo/src/main.c

```diff
@@ -41,9 +41,11 @@ void test_getaddrinfo_no_host(void)
 
 void test_main(void)
 {
+	k_thread_system_pool_assign(k_current_get());
```

Comment:
Does this do anything?  ztest runs the suite in a separate thread, changing the malloc pool for this thread is a noop, right?

Reply:
yes, it does.
It assigns the system heap to the thread resource pool of the main thread.
Child threads, including all the ztest threads, inherit this assignment.